packages/smooth_app/lib/pages/product/product_page/header/product_page_tabs.dart

```diff
@@ -0,0 +1,258 @@
+    required this.labelBuilder,
+    required this.builder,
+  });
+
+  final String id;
+  final String Function(BuildContext) labelBuilder;
+  final Widget Function(BuildContext, Product) builder;
+}
+
+class ProductPageTabBar extends StatelessWidget {
+  const ProductPageTabBar({
+    required this.tabController,
+    required this.tabs,
+  });
+
+  final TabController tabController;
+  final List<ProductPageTab> tabs;
+
+  @override
+  Widget build(BuildContext context) {
+    return SliverPersistentHeader(
+      delegate: _TabBarDelegate(
+        PreferredSize(
+          preferredSize: const Size.fromHeight(SmoothTabBar.TAB_BAR_HEIGHT),
+          child: SmoothTabBar<ProductPageTab>(
+            tabController: tabController,
+            items: tabs.map((ProductPageTab tab) {
+              return SmoothTabBarItem<ProductPageTab>(
+                label: tab.labelBuilder(context),
+                value: tab,
+              );
+            }).toList(growable: false),
+            onTabChanged: (_) {},
+          ),
+        ),
+      ),
+      pinned: true,
+    );
+  }
+
+  /// Keeping this as it will be useful when we re-impletment the reordering
+  /// of tabs in the future.
+  /* void _showReorderBottomSheet(
+    BuildContext context,
+    List<ProductPageTab> tabs,
+  ) {
+    showSmoothReorderBottomSheet<ProductPageTab>(
+      context,
+      items: tabs.map((ProductPageTab tab) {
+        return tab;
+      }).toList(growable: false),
+      onReorder: (List<ProductPageTab> reorderedItems) {
+        context
+            .read<UserPreferences>()
+            .setProductPageTabs(reorderedItems.map((ProductPageTab tab) {
+              return tab.id;
+            }).toList(growable: false));
+        tabs
+          ..clear()
+          ..addAll(reorderedItems);
+      },
+      labelBuilder: (
+        BuildContext context,
+        ProductPageTab item,
+        int index,
+      ) {
+        return Text(
+          item.labelBuilder(context),
+          style: const TextStyle(
+            fontWeight: FontWeight.bold,
+          ),
+        );
+      },
+      title: AppLocalizations.of(context).product_page_reorder_tabs,
+    );
+  } */
+
+  static List<ProductPageTab> extractTabsFromProduct({
+    required BuildContext context,
+    required Product product,
+  }) {
+    final List<ProductPageTab> tabs = <ProductPageTab>[];
+
+    final List<KnowledgePanelElement> roots =
+        KnowledgePanelsBuilder.getRootPanelElements(product);
+    for (final KnowledgePanelElement root in roots) {
+      final String? id = root.panelElement?.panelId;
+      if (id == null) {
+        continue;
+      }
+
+      List<Widget> children = KnowledgePanelsBuilder.getChildren(
+        context,
+        panelElement: root,
+        product: product,
+        onboardingMode: false,
+      );
+
+      final KnowledgePanelTitle knowledgePanelTitle =
+          children.first as KnowledgePanelTitle;
+
+      children = children.sublist(1);
```

Comment:
Could you add a check `if children.size > 1`?
Or… are you sure there always are at least 2 elements?